include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,384 @@
+  * no side effect to sensor instances.
+  *
+  * @param **info For receiving sensor instances' information array pointer.
+  *
+  * @return Number of sensor instances, 0 returned if no sensors.
+  */
+int senss_get_sensors(const struct senss_sensor_info **info);
+
+
+/**
+ * @brief Open sensor instance by sensor type and sensor index.
+ *
+ * Application clients use it to open a sensor instance and get its handle.
+ * Support multiple Application clients for open same sensor instance,
+ * in this case, the returned handle will different for different clients.
+ *
+ * @param type The sensor type which need to open.
+ *
+ * @param sensor_index The index which need to open.
+ *
+ * @param *handle The opened instance handle, if failed will be set to -1.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_open_sensor(
+		int type, int sensor_index,
+		int *handle);
+
+/**
+ * @brief Close sensor instance.
+ *
+ * @param handle The sensor instance handle need to close.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_close_sensor(
+		int handle);
+
+
+/**
+ * @brief Get current report interval of given client handle.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param value The data buffer to receive value (micro seconds unit).
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_get_interval(
+		int handle,
+		uint32_t *value);
+
+/**
+ * @brief Set current report interval of given client handle.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param value The value to be set (micro seconds unit).
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_set_interval(
+		int handle,
+		uint32_t value);
+
+/**
+ * @brief Set current data change sensitivity of given client handle.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support set separated sensitivity for each data field, or global
+ * sensitivity for all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The value to be set.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_set_sensitivity(
+		int handle,
+		int index, uint32_t value);
+
+/**
+ * @brief Get current data change sensitivity of given client handle.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support get separated sensitivity for each data field, or global
+ * sensitivity for all data fields.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The data buffer to receive value.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+int senss_get_sensitivity(
```

Comment:
Same as interval, the handle makes no difference

Reply:
Same as above.

Reply:
I'm going to mark this as resolve in order to keep as few conversation threads open as possible.